### PR TITLE
Fix WooPay duplicated Save my info section

### LINKED ISSUE #9410: WooPay can duplicate the "Save my info" element and AJAX requests on the Checkout Block while Stripe for WooCommerce is active
### Describe the bug

> [!IMPORTANT]
> It's not uncommon for sites to have both WooPayments and [Stripe for WooCommerce](https://woocommerce.com/products/stripe/) active simultaneously.

If a site...
1. Has Stripe for WooCommerce and WooPayments both active simultaneously
2. Has a US-based WooPayments account (which means WooPay will be automatically enabled)
3. Uses the Checkout Block

...and the Stripe for WooCommerce card payment element loads _before_ WooPayments' card payment element, this can:

- Cause `?wc-ajax=wc_stripe_save_appearance` (added in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3047) to load ad infinitum.
- Create a bunch of duplicate "Save my info" elements if/when switching to the WooPayments payment method.

### To Reproduce

1. Onboard WooPayments as a US account (sandbox is fine).
2. Install Stripe for WooCommerce.
3. Activate Stripe for WooCommerce.
4. Configure Stripe for WooCommerce (test mode is fine).
5. Navigate to **WooCommerce > Settings > Payments**.
6. Rearrange the payment methods so that Stripe is at the top of the list.
> [!NOTE]
> This isn't necessary, but it makes testing a _whole_ lot easier.
7. Save your settings.
8. Navigate to **Payments > Settings**.
9. Ensure WooPay is enabled.
10. Add a product to your cart.
11. Clear your server cache
> [!TIP]
> I used [wp cache flush](https://developer.wordpress.org/cli/commands/cache/flush/) via WP-CLI.
12. Navigate to the Checkout Block page.
13. Note that the `Place order` button keeps getting lower on the page.
14. Open up the network requests.
15. Note that `?wc-ajax=wc_stripe_save_appearance` requests are repeating.
16. Select the WooPayments card payment method option.
17. Note that there are now several "Save my info" elements (`woopay-save-new-user-container wc-version-greater-than-91` DIVs) on the page. 
18. Repeat step 8.
19. Disable WooPay.
20. Repeat steps 11–12.
21. Note that the issue is no longer present.

### Actual behavior

When WooPay is enabled but the Stripe for WooCommerce card payment elements load first, this can result in duplicate AJAX requests on the Checkout Block page and create duplicate "Save my info" boxes.

### Screenshots

Here's a video that's too big for GitHub: U01TTK2RXJB/F07LZVCGN7K-slack-file

### Expected behavior

WooPay does not duplicate AJAX calls or elements if WooPayments' card payment element is not the first to load on the checkout page.

### SSR

<details><summary>System Status Report</summary>

```
### WordPress Environment ###

WordPress address (URL): [Redacted]
Site address (URL): [Redacted]
WC Version: 9.2.3
Legacy REST API Package Version: The Legacy REST API plugin is not installed on this site.
Action Scheduler Version: ✔ 3.8.1
Log Directory Writable: ✔
WP Version: 6.6.1
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: ✔

### Server Environment ###

Server Info: nginx
PHP Version: 8.3.11
PHP Post Max Size: 2 GB
PHP Time Limit: 1200
PHP Max Input Vars: 6144
cURL Version: 8.9.1
OpenSSL/3.0.14

SUHOSIN Installed: –
MySQL Version: 10.6.17-MariaDB-log
Max Upload Size: 2 GB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

[REDACTED]

### Post Type Counts ###

attachment: 1
page: 8
pos_outlet: 1
pos_receipt: 1
pos_register: 1
pos_session: 1
product: 4
revision: 6
shop_order: 58
shop_order_placehold: 15
shop_order_refund: 5
shop_subscription: 2
wp_font_face: 33
wp_font_family: 12
wp_global_styles: 1
wp_navigation: 1
wp_template: 2
wp_template_part: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (5) ###

Jetpack: by Automattic – 13.8
WooCommerce.com Update Manager: by Automattic – 1.0.3
WooCommerce Stripe Gateway: by WooCommerce – 8.6.1
WooPayments: by WooCommerce – 8.1.1
WooCommerce: by Automattic – 9.2.3

### Inactive Plugins (6) ###

Akismet Anti-spam: Spam Protection: by Automattic - Anti-spam Team – 5.3.3
Checkout Plugins - Stripe for WooCommerce: by Checkout Plugins – 1.10.0
Code Snippets: by Code Snippets Pro – 3.6.5.1
Payment Plugins for Stripe WooCommerce: by Payment Plugins
support@paymentplugins.com – 3.3.75

WooCommerce Subscriptions: by WooCommerce – 6.5.0 (update to version 6.6.2 is available)
WP Console: by Edi Amin – 2.4.1

### Dropin Plugins () ###

advanced-cache.php: advanced-cache.php
object-cache.php: Memcached

### Settings ###

Legacy API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left_space
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: bundle (bundle)
external (external)
grouped (grouped)
simple (simple)
subscription (subscription)
variable (variable)
variable subscription (variable-subscription)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: ✔
Enforce Approved Product Download Directories: ✔
HPOS feature enabled: ✔
Order datastore: Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore
HPOS data sync enabled: –

### Logging ###

Enabled: ✔
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 30 days
Level threshold: –
Log directory size: 801 KB

### WC Pages ###

Shop base: #5 - /shop/
Cart: #6 - /cart/ -  Contains the woocommerce/cart block
Checkout: #7 - /checkout/ -  Contains the woocommerce/checkout block
My account: #8 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Twenty Twenty-Four
Version: 1.2
Author URL: https://wordpress.org
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ❌ Not declared

### Templates ###

Overrides: /srv/htdocs/wp-content/plugins/woocommerce/templates/block-notices/error.php
/srv/htdocs/wp-content/plugins/woocommerce/templates/block-notices/notice.php
/srv/htdocs/wp-content/plugins/woocommerce/templates/block-notices/success.php


### WooPayments ###

Version: 8.1.1
Connected to WPCOM: Yes
WPCOM Blog ID: 230681530
Account ID: acct_1PfQLPCImV0VIQ30
Payment Gateway: Enabled
Test Mode: Enabled
Enabled APMs: card
WooPay: Enabled (product,cart,checkout)
WooPay Incompatible Extensions: No
Apple Pay / Google Pay: Enabled (product,cart,checkout)
Fraud Protection Level: basic
Multi-currency: Disabled
Auth and Capture: Enabled
Documents: Disabled
Logging: Enabled

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customize-store
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
pattern-toolkit-full-composability
product-custom-fields
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
printful
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page
launch-your-store

Disabled Features: experimental-blocks
minified-js
product-pre-publish-modal
settings
async-product-editor-category-field
product-editor-template-system

Daily Cron: ✔ Next scheduled: 2024-09-09 21:44:22 +00:00
Options: ✔
Notes: 66
Onboarding: skipped

### Action Scheduler ###

Complete: 174
Oldest: 2024-08-10 07:22:48 +0000
Newest: 2024-09-09 18:14:15 +0000

Failed: 79
Oldest: 2024-03-18 23:44:21 +0000
Newest: 2024-08-31 17:11:47 +0000

Pending: 1
Oldest: 2024-09-10 07:31:00 +0000
Newest: 2024-09-10 07:31:00 +0000


### Status report information ###

Generated at: 2024-09-09 18:59:04 +00:00
```
</details>

### Additional context

p1725645460123609-slack-CHG7MTCAF

### PR DESCRIPTION
**Pull request by alefesouza**
Fixes #9410

#### Changes proposed in this Pull Request

This PR fixes the WooPay `Save my info` component being duplicated due to late ajax requests before the payment method component is fully rendered, it was being duplicated [here](https://github.com/Automattic/woocommerce-payments/blob/319de19d321d5802b94b5dadfeee480b8f7bd643/client/checkout/woopay/index.js#L89).

The issue mentions it's related to Stripe plugin but it happens even without the Stripe plugin installed.

#### Testing instructions

You can try to reproduce this bug without checking out to the branch:

* Add a product to the cart.
* Duplicate [this line](https://github.com/Automattic/woocommerce-payments/blob/319de19d321d5802b94b5dadfeee480b8f7bd643/client/checkout/woopay/index.js#L83).
* Access blocks checkout page.
* The WooPay `Save my info` component should be duplicated.

Screenshot 2024-10-06 at 17 17 00

* Checkout to this PR.
* The WooPay `Save my info` component should not duplicate anymore.

This PR fixes that by checking if the element where the component is mounted already exists on the `renderSaveUserSection` before creating a new element for it.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9531` or branch name `fix/woopay-duplicated-component` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/woopay-duplicated-component) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 27706b808e90c042241aecf38a30bc4a32ea1b55
- Build time: 2024-10-22 13:13:08 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Comment by asumaran** [comment]
@alefesouza, I encountered the bug you’re fixing here through a different scenario. I pulled your branch and confirmed that this PR resolves the issue as well.

I’m leaving the steps to reproduce the bug I encountered for future reference. It would be great if you could confirm the fix on your end too.

Steps to reproduce:

1. Add a product that requires shipping to the cart.
2. Go to the checkout page and enter a shipping address (make sure shipping options are available).
3. Navigate to WooCommerce > Settings.
4. Set “Shipping location(s)” to “Disable shipping & shipping calculations.”
5. Refresh the checkout page.
6. You should see a duplicated “Save my info” section.
7. With this branch, the issue should no longer occur.

File: client/checkout/woopay/index.js [hunk 1]
```diff
@@ -23,31 +23,38 @@ const renderSaveUserSection = () => {
...
+				paymentOptions.parentNode.insertBefore(
+					checkoutPageSaveUserContainer,
+					paymentOptions.nextSibling
+				);
+
+				paymentOptions.classList.add( 'is-woopay' );
+			}
 		}
+
+		ReactDOM.render(
```
Comment (a-danae) [inline client/checkout/woopay/index.js:54]: Things behave as expected, but I'm curious. Why keep this render when the element already exists?

**Review by a-danae — APPROVED** [review]
Things work as expected in this branch 👍 

I was able to replicate the problem in `trunk` using the Stripe extension. The "Save my info" component was inserted multiple times on the page. This behavior was caused by [this listener](https://github.com/Automattic/woocommerce-payments/blob/develop/client/checkout/woopay/index.js#L79-L84) that inserts the component when an AJAX request is completed.

With this approach, there's still an event triggered on AJAX requests, but now the element is only inserted if it doesn't already exist.

✅ The "Save my info" component isn't rendered multiple times on the blocks checkout.
✅ The "Save my info" component continues to be displayed as expected on the shortcode checkout.
✅ The WooPay user gets created in both the shortcode and blocks checkout when checking off "Save my info"

I left a question below but it's not a blocker.

I noticed the following behavior but it's also happening on `develop`. So I'm just noting it down and I'll create a separate issue for it.

Sometimes the "Save my info" checkbox is displayed, even though the user already exists in WooPay. Seems to be related to the ajax check for whether the user exists not being triggered sometimes. The checkbox is always hidden when hard refreshing the page. So might be related to caching. Let's dive into it in another issue.

| Checkbox is hidden | Checkbox is displayed |
|--------|--------|
|  |  | 

Thanks for the work here @alefesouza ! And apologies for the delay. I'm merging this since you're AFK.

**Pull request merged**